File: Data Structures & Algorithms/three-integer-sum/submission-1.cs
public class Solution {
    public List<List<int>> ThreeSum(int[] nums) {

        List<List<int>> res = new();
        HashSet<string> tupleSeen = new();
        HashSet<int> numSeen = new();

        for(int i = 0; i < nums.Length; i++)
        {
            for(int j = i+1; j < nums.Length; j++)
            {
                int num = -nums[i]-nums[j];
                List<int> temp = new List<int> {nums[i], nums[j], num};
                temp.Sort();
                string key = string.Join(",", temp);
                
                if(numSeen.Contains(num) && !tupleSeen.Contains(key))
                {
                    res.Add(new List<int> {nums[i], nums[j], num});
                    tupleSeen.Add(key);
                }
            }
            numSeen.Add(nums[i]);
        }
        return res;
    }
}

File: Data Structures & Algorithms/jump-game/submission-1.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        
        for(int i = 0, j = 0; i < nums.size() && i <= j; i++)
        {
            j = max(j, i+nums[i]);
            if(i == nums.size()-1) return true;
        }
        return false;
    }
};
